client/my-sites/themes/helpers.js

```diff
@@ -162,7 +163,12 @@ export function interlaceThemes( wpComThemes, wpOrgThemes, searchTerm, isLastPag
 	const restWpComThemes = matchingTheme
 		? wpComThemes.filter( ( theme ) => theme.id !== matchingTheme.id )
 		: wpComThemes;
-	interlacedThemes.push( ...restWpComThemes );
+
+	// The themes endpoint returns retired themes when search term exists.
+	// See: https://github.com/Automattic/wp-calypso/pull/78231
+	interlacedThemes.push(
+		...( searchTerm ? restWpComThemes.filter( ( theme ) => ! theme.retired ) : restWpComThemes )
```

Comment:
[nit] We can filter out the retired themes regardless of the search term. But the current way is also fine 👍

Reply:
I'll leave the logic in for now, since I like the idea of the small performance gain! 🙂